### PR TITLE
Use `Schedulers#immediate` as a default scheduler

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java [hunk 1]
```diff
@@ -238,26 +238,30 @@ public void run() {
             try {
                 if (commitArgs != null) {
                     if (!commitArgs.offsets().isEmpty()) {
-                        inProgress.incrementAndGet();
                         switch (ackMode) {
                             case ATMOST_ONCE:
                                 consumer.commitSync(commitArgs.offsets());
```
Comment (bsideup) [inline src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java:243]: Synchronous, no need to increment/decrement I assume

File: src/test/java/reactor/kafka/receiver/internals/MockReceiverTest.java [hunk 2]
```diff
@@ -596,15 +596,17 @@ public void manualAckCommitBatchSize() {
...
-            long offset = r.receiverOffset().offset();
-            if (offset < 10) {
-                r.receiverOffset().acknowledge();
-                if (((offset + 1) % batchSize) == 0)
-                    lastCommitted.set(offset);
-            } else
-                uncommittedMessages.add(r);
-            verifyCommit(r, lastCommitted.get());
-            return Mono.just(r);
+            return Mono.fromCallable(() -> {
```
Comment (bsideup) [inline src/test/java/reactor/kafka/receiver/internals/MockReceiverTest.java:599]: I spent ~2 days debugging the tests after applying that small change.

"never block the emitter thread" makes a lot more sense now 😂

File: src/test/java/reactor/kafka/sender/internals/MockTransactionTest.java [hunk 3]
```diff
@@ -133,10 +133,10 @@ public void transactionalReceiveAndSend() throws Exception {
...
 
-        assertEquals(transactionCount, producer.beginCount);
-        assertEquals(transactionCount, producer.commitCount);
-        assertEquals(0, producer.abortCount);
-        assertEquals(transactionCount, producer.sendOffsetsCount);
+        assertThat(producer.beginCount).as("beginCount").isGreaterThanOrEqualTo(transactionCount);
+        assertThat(producer.commitCount).as("commitCount").isGreaterThanOrEqualTo(transactionCount);
+        assertThat(producer.abortCount).as("abortCount").isZero();
+        assertThat(producer.sendOffsetsCount).as("sendOffsetsCount").isEqualTo(transactionCount);
     }
```
Comment (garyrussell) [inline src/test/java/reactor/kafka/sender/internals/MockTransactionTest.java:140]: @bsideup This should fix the flakiness - yes, it's possible you will get less then 10 per poll.

Unfortunately, there is no `poll.min.records`, you can only approximate it with `fetch.min.bytes`.

Comment (bsideup) [reply src/test/java/reactor/kafka/sender/internals/MockTransactionTest.java:140]: Thanks! This is the assumption I had but wasn't sure about, appreciate the clarification 👍

**Pull request merged**